### ISSUE TITLE
Add Test Coverage to Selling of Positions

### ISSUE THREAD

**Issue by pwdel** [openpredictionmarkets/socialpredict#291]
After this is completed:

https://github.com/orgs/openpredictionmarkets/projects/4?pane=issue&itemId=86908606&issue=openpredictionmarkets%7Csocialpredict%7C406

Add Test Coverage to Selling of Positions

Some thoughts:

* Presuming we only sell *owned* positions:
* Let's assume we have a user that bought a lot (100) X, and then their position becomes 0 because so many others bought 1000's of Y forcing the price/probability very far towards the Y direction.
* Now this user buys 10 Y.
* There we never a force sale of 100 of X in order to buy that 10 Y, so hypothetically if the market swings the other direction, they can still benefit from that 100 X position if the market swings the other way. This is a problem because we're not really forcing that user out of their original X position.
* It seems like we should force sell all previous X positions with zero profit and DON'T reimburse the user's wallet.

---

* On the other hand, if we force sale all CURRENCY out of a market, this lowers the market volume, discouraging future participation and increasing volatility.
* Position is a different thing than, "amount spent," on a market. Position is different than, "profit," as someone may play two sides of a market heavily and not win as much due to one's position...which is just an unfortunate part of being wrong and not playing a market correctly or being reactive amount, from the standpoint of that user.
* We want to incentivize people to uncover binary truths where possible which means, we want to be able to bury / punish investors in the past who were wrong, hence the, "tax on bullshit."
* So basically, to align with what we're getting at, we can allow someone's position to be force sold before buying in an opposite direction, but we can't allow an individual to withdraw their previous investment...it becomes a neutral bet from their standpoint and they will get a portion of that in the future if the market now resolves in their favor.
* So from that sense, it also encourages switching sides, e.g. at least you will be able to recuperate some of your past incorrect investment. It may also encourage selling off one's incorrect investment if an external truth becomes true, which encourages prices to be set more correctly.
